### ISSUE TITLE
bags leaving gaps when sorting and bizarre behavior

### ISSUE THREAD

**Issue by Kheath812** [Jaliborc/Bagnon#630]
So, my bags are behaving bizarrely. I'm trying to sort them and the sorting function keeps leaving gaps in my bags.
![asdsaad](https://user-images.githubusercontent.com/8062471/34086595-97497f34-e362-11e7-8b2c-1f24873d658a.png)

As you can see, i dont have reverse bag cleanup on and i dont have reverse bag order on and items are still sorting to the bottom. now, i **want** my bags to sort to the bottom right, so thats fine, but why is it happening when i dont have the option checked for it to do that? it's at odds with my settings.

Secondly, if my items are sorting in reverse, why does it have a giant gap in my middle bag? the middle bag is not reverse sorting but the rightmost 2 bags are? what the shit is going on lol.

Third, If i turn on loot to leftmost bag, it loots new items to my middle bag (the orange x slot), which doesnt even make sense. it's not my left _or_ my rightmost bag if i somehow had it reversed, so why is it even looting to that bag to begin with? If it's turned off, I loot to the top left slot as normal (the brown satchel). I dont have any bags ignored. Why is it looting to my leftmost bag with the Loot to Leftmost bag option turned _OFF_, but not _ON_??

I've tried every combination of settings I can think of, including the commands

/run SetInsertItemsLeftToRight(true/false)
and
/run SetSortBagsRightToLeft(true/false)

and I can't seem to fix it. I know a lot of peoples problem is some combination of those commands, but i've already tried that. Am i just missing something completely?

**Comment by Kheath812** [comment]
for reference, here's the sorting on default bags. it's all reverse sorting properly, no gaps.
![dfsfdsf](https://user-images.githubusercontent.com/8062471/34087114-a17eede6-e366-11e7-8fb4-399b2c88edf4.png)

**Comment by PandarinDev** [comment]
Go into Bagnon options and make sure that "Reverse Bag Order" and "Reverse Slot Order" are both unchecked. Click sort, try to loot something and validate if everything works to your expectations.

Now configure your sorting/looting behaviour to your expectations with the two functions you mentioned. Voilá.

**Comment by Kheath812** [comment]
as per my first picture, you can see that those options are already off and it still leaves the gap when sorting

**Comment by PandarinDev** [comment]
No, on your first picture you are showing off advanced interface options. I was talking about Bagnon options. Open your backpack and click on the gear icon or select Bagnon on the same window where you selected AdvancedInterfaceOptions.

**Comment by Kheath812** [comment]
the reverse bag order and reverse sort order are from bagnon options, i just copy pasted it into the screenshot.

**Comment by kylewebdev** [comment]
If you right-click on a bag and set the "Assign to Bag:" option to the same on all your bags like the "Equipment" option, this will fix the issue. I had the same thing. This is caused by the default sorting mechanic.

**Comment by richlira** [comment]
Well, suddenly one of my chars started this behavior.

I'm using Bagnon on Retail and Classic.

The issue appeared on one of my chars in Retail.

Anyone found a solution?
